pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -329,7 +328,7 @@ func resolveStringWithTemplateVars(ctx context.Context, in string, svc listeners
 		if k == "host" {
 			adHocTemplateVars[k] = func(ctx context.Context, tplVar string, svc listeners.Service) (string, error) {
 				host, err := getHost(ctx, tplVar, svc)
-				if ipv6Re.MatchString(host) {
+				if apiutil.IsIPv6(host) {
```

Comment:
Just a question since my knowledge of AD is limited: There is a fundamental change here that the new method will work correctly only on pure IP addresses and bare hostnames but I see that the comment might indicate a bit more might be included in the `host` var (maybe port). Are we sure that we will only get IPs in that var?

Reply:
The `%host%` autodiscovery template variable should normally be substituted by the IP of the discovered container. Either an IPv4 or an IPv6.
The previous code was working fine in those two nominal cases.

However, there are corner cases where `%host%` is substituted with a hostname instead of an IP: https://github.com/DataDog/datadog-agent/blob/596163c65d7b84a0d7779438873abe54250d1c18/pkg/autodiscovery/listeners/container.go#L136-L141

The previous code was broken in case `%host%` was substituted with a hostname composed of only hexadecimal digits. In this case, it was matching the regular expression although it wasn’t a valid IPv6.

The thing is that [the regular expression that matches IPv6 and only IPv6](https://stackoverflow.com/a/17871737) is so complex that it’s the sign that regex are not the good tool for that job.